components/api-server/src/client.rs

```diff
@@ -0,0 +1,265 @@
+use clp_rust_utils::{
+    clp_config::package::{
+        config::{Config, StorageEngine},
+        credentials::Credentials,
+    },
+    database::mysql::create_mysql_pool,
+    job_config::{QUERY_JOBS_TABLE_NAME, QueryJobStatus, QueryJobType, SearchJobConfig},
+};
+use futures::{Stream, StreamExt};
+use num_enum::TryFromPrimitive;
+use serde::{Deserialize, Serialize};
+use sqlx::Row;
+use thiserror::Error;
+
+#[derive(Clone)]
+pub struct Client {
+    #[allow(clippy::struct_field_names)]
+    mongodb_client: mongodb::Client,
+    sql_pool: sqlx::Pool<sqlx::MySql>,
+    config: Config,
+}
+
+impl Client {
+    /// Establishes a connection to both the `MySQL` database and the `MongoDB` database,
+    /// creating a new `Self` (Client) instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - A `Config` struct containing database and cache configuration details.
+    /// * `credentials` - A `Credentials` struct containing sensitive database login information.
+    ///
+    /// # Errors
+    ///
+    /// Returns a `ClientError` if either the `MySQL` connection pool creation or
+    /// the `MongoDB` client initialization fails. This includes networking issues,
+    /// invalid credentials, or incorrect configuration settings.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` which is `Ok(Self)` (the connected `Client` instance) on success,
+    /// or `Err(ClientError)` on failure.
```

Comment:
```suggestion
    /// Factory method to create a new client with active connections to both MySQL and MongoDB
    /// databases.
    ///
    /// # Returns
    ///
    /// A newly created [`Client`] instance with active connections to both databases.
    ///
    /// # Errors
    ///
    /// Returns an error if:
    ///
    /// * Forwards [`create_mysql_pool`]'s errors on failure.
    /// * Forwards [`mongodb::Client::with_uri_str`]'s errors on failure.
```
To make it consistent with other docstrings. Key changes to follow our guidelines:
- Move `returns` section before `errors`.
- Drop the arg section. I think Rust APIs usually don't document the arguments to encourage self-explanatory code.
- Use `[]` to surround symbols. This allows IDEs to render a hyperline to the corresponding symbol.
- In `Returns` section, we only explain what's returned on success.
- In `Errors` section, we document any propagated errors using `Forwards xxx's return values` format.